File: app/src/main/java/torille/fi/lurkforreddit/data/models/view/PostAndComments.java
package torille.fi.lurkforreddit.data.models.view;

import com.google.auto.value.AutoValue;

import java.util.List;

/**
 * Created by eva on 2.5.2017.
 */
@AutoValue
public abstract class PostAndComments {

    public abstract Post originalPost();

    public abstract List<Comment> comments();

    public static PostAndComments create(Post post, List<Comment> comments) {
        return new AutoValue_PostAndComments(post, comments);
    }

}

File: app/src/main/java/torille/fi/lurkforreddit/data/models/view/PostAndComments.kt
package torille.fi.lurkforreddit.data.models.view

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

/**
 * Class that holds holds comments and its post
 */
@Parcelize
data class PostAndComments(
        val originalPost: Post,
        val comments: List<Comment>
) : Parcelable
